lms/djangoapps/teams/views.py

```diff
@@ -326,37 +342,45 @@ def get(self, request):
             result_filter.update({'topic_id': request.QUERY_PARAMS['topic_id']})
         if 'include_inactive' in request.QUERY_PARAMS and request.QUERY_PARAMS['include_inactive'].lower() == 'true':
             del result_filter['is_active']
-        if 'text_search' in request.QUERY_PARAMS:
-            return Response(
-                build_api_error(ugettext_noop("text_search is not yet supported.")),
-                status=status.HTTP_400_BAD_REQUEST
-            )
 
         queryset = CourseTeam.objects.filter(**result_filter)
 
-        order_by_input = request.QUERY_PARAMS.get('order_by', 'name')
-        if order_by_input == 'name':
-            queryset = queryset.extra(select={'lower_name': "lower(name)"})
-            queryset = queryset.order_by('lower_name')
-        elif order_by_input == 'open_slots':
-            queryset = queryset.annotate(team_size=Count('users'))
-            queryset = queryset.order_by('team_size', '-last_activity_at')
-        elif order_by_input == 'last_activity_at':
-            queryset = queryset.annotate(team_size=Count('users'))
-            queryset = queryset.order_by('-last_activity_at', 'team_size')
+        if 'text_search' in request.QUERY_PARAMS and \
```

Comment:
You could also pull out `'text_search' in request.QUERY_PARAMS` into a variable.